src/ansys/fluent/visualization/contour.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Optional
+
+
+class Contour:
+    def error_check(self, solver):
+        allowed_fields = (
+            solver.field_data.get_scalar_field_data.field_name.allowed_values()
+        )
+        allowed_surfaces = (
+            solver.field_data.get_scalar_field_data.surface_name.allowed_values()
+        )
+        if self.field not in allowed_fields:
+            raise ValueError(
+                f"{self.field} is not valid field. Valid fields are - {allowed_fields}"
```

Comment:
The hyphens are wrong in these sentences.

Reply:
We can use the FieldInfo class. In fact it would be good to have a validate methods on FieldInfo to avoid repeating boilerplate. Note that in field_data,py, we use exceptions that encapsulate these errors quite nicely.

Reply:
> The hyphens are wrong in these sentences.

Done. Thank you.

Reply:
> We can use the FieldInfo class. In fact it would be good to have a validate methods on FieldInfo to avoid repeating boilerplate. Note that in field_data,py, we use exceptions that encapsulate these errors quite nicely.

field_info returns dict in both field and surface methods and we need to extract data from it which is redundant.

field_data returns list and it is convenient to use it directly. We just need names in this case.

![image](https://github.com/ansys/pyfluent-visualization/assets/106588300/b76b4b4e-aae9-4ae4-bfc3-cfde06363c7f)